src/client.rs

```diff
@@ -1912,31 +1954,107 @@ impl TryFrom<&ChallengeRef<'_>> for BearerChallenge {
     }
 }
 
-/// Extract `Docker-Content-Digest` header from manifest GET or HEAD request.
-/// Can optionally supply a response body (i.e. the manifest itself) to
-/// fallback to manually hashing this content. This should only be done if the
-/// response body contains the image manifest.
-fn digest_header_value(headers: HeaderMap, body: Option<&[u8]>) -> Result<String> {
-    let digest_header = headers.get("Docker-Content-Digest");
-    match digest_header {
-        None => {
-            if let Some(body) = body {
-                // Fallback to hashing payload (tested with ECR)
-                let digest = sha2::Sha256::digest(body);
-                let hex = format!("sha256:{:x}", digest);
-                debug!(%hex, "Computed digest of manifest payload.");
-                Ok(hex)
-            } else {
-                Err(OciDistributionError::RegistryNoDigestError)
+/// Helper for extracting `Docker-Content-Digest` header from manifest GET or HEAD request.
+fn digest_header_value(headers: HeaderMap) -> Result<Option<String>> {
+    headers
+        .get(DOCKER_DIGEST_HEADER)
+        .map(|hv| hv.to_str().map(|s| s.to_string()))
+        .transpose()
+        .map_err(|e| {
+            OciDistributionError::GenericError(Some(format!(
+                "Cannot convert Docker-Content-Digest header value to string: {}",
+                e
+            )))
+        })
+}
+
+/// Given the optional digest header value and digest of the reference, returns the digest of the
+/// content, validating that the digest of the content matches the proper digest
+fn validate_digest(
```

Comment:
Dropping the comment here, but it applies to the other functions introduced by this PR. I feel `client.rs` is growing too much, I would like to split that into smaller modules to improve maintenance.
While this should be done with a future PR, maybe these functions should already be placed inside of an internal module.

Reply:
Yeah this is why I created the new `blob.rs` file. I can break these out too